### PR TITLE
smt2 bit vector fix [blocks: any PR adding CBMC tests]

### PR DESCRIPTION
**Pull request by kroening**
This is #3119 plus a commit that enables the SMT2 tests that subsequently pass.

- [ ] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [X] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- [X] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: scripts/delete_failing_smt2_solver_tests [hunk 1]
```diff
@@ -211,107 +131,57 @@ rm equality_through_array_of_struct1/test.desc
...
-rm switch2/test.desc
-rm switch3/test.desc
-rm switch4/test.desc
-rm switch5/test.desc
-rm trace_address_arithmetic1/test.desc
 rm trace-values/trace-values.desc
-rm trace_options_json_extended/extended.desc
-rm trace_options_json_extended/non-extended.desc
-rm trace_show_code/test.desc
+rm trace_address_arithmetic1/test.desc
```
Comment (martin-cs) [inline scripts/delete_failing_smt2_solver_tests:181]: Does this mean that ```trace_address_arithmetic1``` used to pass but no longer does?

**Review by martin-cs — APPROVED** [review]
Fine modulo the one comment.

File: scripts/delete_failing_smt2_solver_tests (hunk 1 above)
Comment (kroening) [reply scripts/delete_failing_smt2_solver_tests:181]: No, just that the list wasn't properly sorted before (scroll up a bit).

Comment (martin-cs) [reply scripts/delete_failing_smt2_solver_tests:181]: Point.

**Review by chrisr-diffblue — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 33cfe25a4c4dc3a9d9310964ececa10046c033de).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/90470839

**Review by peterschrammel — APPROVED** [review]
I don't quite like the delete script; would prefer adding a tag to the .desc file as done for excluding tests that don't yet run with symex-driven lazy loading, but ok; hope this becomes obsolete soon.

**Comment by kroening** [comment]
@peterschrammel Let's say we'll get the number of non-working tests below 50, and then we switch to the flag!

**Pull request merged**